src/state/store.js

```diff
@@ -160,31 +162,38 @@ export const hasDefaultControlFields = derived(store, ($store) =>
 // ///// probe querying infrastructure.
 
 function getParamsForQueryString(obj) {
-  return {
-    versions: obj.versions,
-    channel: obj.channel,
-    os: obj.os,
-    aggregationLevel: obj.aggregationLevel,
-    process: obj.process,
-    timeHorizon: obj.timeHorizon,
-    proportionMetricType: obj.proportionMetricType,
-    activeBuckets: obj.activeBuckets,
-    visiblePercentiles: obj.visiblePercentiles,
-  };
+  if (obj.product === 'firefox') {
+    return {
+      channel: obj.productDimensions.channel,
+      os: obj.productDimensions.os,
+      aggregationLevel: obj.productDimensions.aggregationLevel,
+      process: obj.productDimensions.process,
+      timeHorizon: obj.timeHorizon,
+      proportionMetricType: obj.proportionMetricType,
+      activeBuckets: obj.activeBuckets,
+      visiblePercentiles: obj.visiblePercentiles,
+    };
+  }
+  throw Error('Product not recognized.');
 }
 
 function getParamsForDataAPI(obj) {
-  const channelValue = getFieldValueKey('channel', obj.channel);
-  const osValue = getFieldValueKey('os', obj.os);
-  const params = getParamsForQueryString(obj);
-  delete params.timeHorizon;
-  delete params.proportionMetricType;
-  delete params.activeBuckets;
-  delete params.visiblePercentiles;
-  params.probe = obj.probeName;
-  params.os = osValue;
-  params.channel = channelValue;
-  return params;
+  if (obj.product === 'firefox') {
+    const channelValue = getFieldValueKey('channel', obj.productDimensions.channel);
+    const osValue = getFieldValueKey('os', obj.productDimensions.os);
+    const process = getFieldValueKey('os', obj.productDimensions.process);
+    const params = getParamsForQueryString(obj);
+    delete params.timeHorizon;
+    delete params.proportionMetricType;
+    delete params.activeBuckets;
+    delete params.visiblePercentiles;
+    params.probe = obj.probeName;
+    params.os = osValue;
+    params.channel = channelValue;
+    params.process = process;
+    params.versions = []; // TODO: This should not be expected by the API.
```

Comment:
@robhudson we found that this param is still expected by the API, probably something to flag for future removal

Reply:
@spasovski let's file an issue for this so we can track that separately.